File: Samples for Solid Edge/[VB]-Samples for Solid Edge/VB/SolidEdge.ApplicationEvents/ControlExtensions.vb
Imports System.Text

Namespace SolidEdge.ApplicationEvents
	Public Module ControlExtensions
		<System.Runtime.CompilerServices.Extension> _
		Public Sub [Do](Of TControl As Control)(ByVal control As TControl, ByVal action As Action(Of TControl))
			If control.InvokeRequired Then
				control.Invoke(action, control)
			Else
				action(control)
			End If
		End Sub
	End Module
End Namespace

File: Samples for Solid Edge/[C#]-Samples for Solid Edge/C#/SolidEdge.ApplicationEvents/ControlExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SolidEdge.ApplicationEvents
{
    public static class ControlExtensions
    {
        public static void Do<TControl>(this TControl control, Action<TControl> action)
          where TControl : Control
        {
            if (control.InvokeRequired)
            {
                control.Invoke(action, control);
            }
            else
            {
                action(control);
            }
        }
    }
}
